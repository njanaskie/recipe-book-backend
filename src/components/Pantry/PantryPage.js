import React, { useReducer, useEffect } from 'react';
import IngredientsList from '../Ingredients/IngredientsList'
import IngredientsContext from '../../../context/ingredients-context'
import ingredientsReducer from '../../reducers/ingredients'
import PantryContext from '../../../context/pantry-context'
import pantryReducer from '../../reducers/pantry'

const PantryPage = () => {
    const [ingredients, dispatch] = useReducer(ingredientsReducer, [])
    const [pantryIngredients, pantryDispatch] = useReducer(pantryReducer, [])

    return (
        <PantryContext.Provider value={{ pantryIngredients, pantryDispatch }}>
            <IngredientsContext.Provider value={{ ingredients, dispatch }}>
                <h3>Pantry page content</h3>
                <IngredientsList />
            </IngredientsContext.Provider>
        </PantryContext.Provider>

    )
};

export default PantryPage;