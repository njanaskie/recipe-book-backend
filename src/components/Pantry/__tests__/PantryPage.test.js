import React from 'react'
import { shallow } from 'enzyme'
import PantryPage from '../PantryPage'

test('should render PantryPage correctly',() => {
    const wrapper = shallow(<PantryPage />)
    expect(wrapper).toMatchSnapshot()
})