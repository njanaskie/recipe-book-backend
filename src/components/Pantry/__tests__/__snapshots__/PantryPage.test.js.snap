// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PantryPage correctly 1`] = `
<ContextProvider
  value={
    Object {
      "pantryDispatch": [Function],
      "pantryIngredients": Array [],
    }
  }
>
  <ContextProvider
    value={
      Object {
        "dispatch": [Function],
        "ingredients": Array [],
      }
    }
  >
    <h3>
      Pantry page content
    </h3>
    <IngredientsList />
  </ContextProvider>
</ContextProvider>
`;
